File: P02 - Kelas Objek/Python/Bag. pertama/Kelas06.py
# File: Kelas06.py
# Penulis : Elmosius Suli
# Default arguments & type hint

class Kelas06:
    # Atribut yang dituliskan Langsung didaLam keLas akan menjadt class member
	# Class member akan dipeLajari nanti

    def	__init__(self, __id: int, name: str = "Noname", _age: int = 30) -> None:
        # "-> None" : type hint, method tidak mengembaLikan niLai
        # "id: int, name: str, _age: int" : type hint, tipe data masing2 argument.
        # "name: str = "Noname", _age: int = 30" : default argument, dipakai jika tidak diisi 
        self.__id =	__id	    # private
        self.name = name        # public
        self._age = _age        # protected

    def cetak(self) -> None:    # -› None = type hint, None = method tidak mengembaLikan apa2 
        print("ID:", self.__id, "| Nama:", self.name, "| Umur:", self._age)
        
## Program utama ##
def main():
    k1 = Kelas06(3, "Hudson Taylor", 20)        
    k2 = Kelas06(4)   
   
    # Coba buka comment baris2 kode int:
    # Error karena td private jadi ttdak bisa dtakses dart Luar keLas
    # print("Print gagaL karena ID private")
    # print("ID:", k1. id, "I Nama:", k1.name, "1 Umur:", k1._age)
    # print("ID:", k2. td, "/ Nama:", k2. name, "1 Umur:", k2._age)
     
    # Ini juga bisa kerena method cetak ada didaLam KeLas06
    print("Panggil method cetak()")
    k1.cetak()
    k2.cetak()


    
if __name__ == '__main__':
    main()

File: P02 - Kelas Objek/Java/Bag. pertama/Kelas06.java
package Praktikum.P02;

// Method / constructor Overloading sebagai pengganti default arguments
// karena Java tidak memiliki default arguments
public class Kelas06 {
    // attribut2
    private int id;
    public String name;
    protected int age;


    // Konstuktor
    public Kelas06(int id, String nama, int age){
        this.id   = id;             // this = objek yg sedang diproses
        this.name = nama;
        this.age  = age;
    }

    // Konstuktor 2 (overloading)
    public Kelas06(int id, String nama){
        this(id, nama, 30);     // Panggil konstruktor 2 parameter, df default age = 30
    }

    // Konstuktor 3 (overloading)
    public Kelas06(int id){
        this(id, "Noname", 30);            // Panggil konstruktor 3 parameter, df default name & age
    }

    // Method cetak
    public void cetak(){
        System.out.println("ID: " + this.id + " | Nama: " + this.name + " | Umur: " + this.age);
    }

    public static void main(String[] args){
        Kelas06 k1 = new Kelas06(3, "Hudson Taylor", 20);   //k1 = pbkel daro Kelas06
        Kelas06 k2 = new Kelas06(4, "Nelson Mandela");          //k2 = objeck dari Kelas06 juga
        Kelas06 k3 = new Kelas06(5);                                 //k3 = objeck dari Kelas06 juga

        System.out.println("Cetak dengan method cetak():");
        k1.cetak();
        k2.cetak();
        k3.cetak();
    }

}
